src/xcode/ENA/ENA/Source/Services/HealthCertificateValidationService/HealthCertificateValidationService.swift

```diff
@@ -172,4 +212,557 @@ final class HealthCertificateValidationService: HealthCertificateValidationProvi
 			completion(.failure(.ONBOARDED_COUNTRIES_JSON_DECODING_FAILED))
 		}
 	}
+	
+	// MARK: - Flow
+	
+	private func proceedWithUpdatingValueSets(
+		healthCertificate: HealthCertificate,
+		arrivalCountry: String,
+		validationClock: Date,
+		completion: @escaping (Result<HealthCertificateValidationReport, HealthCertificateValidationError>) -> Void
+	) {
+		// 2. update/ download value sets
+		vaccinationValueSetsProvider.latestVaccinationCertificateValueSets()
+			.sink(
+				receiveCompletion: { result in
+					switch result {
+					case .finished:
+						break
+					case .failure(let error):
+						if case let URLSession.Response.Failure.httpError(_, response) = error {
+							switch response.statusCode {
+							case 500...509:
+								Log.error("Failed to update value sets with status code: \(response.statusCode)", log: .vaccination, error: error)
+								completion(.failure(.VALUE_SET_SERVER_ERROR))
+							default:
+								Log.error("Unhandled Status Code while fetching certificate value sets", log: .vaccination, error: error)
+							}
+							
+						} else if case URLSession.Response.Failure.noNetworkConnection = error {
+							Log.error("Failed to update value sets. No network error", log: .vaccination, error: error)
+							completion(.failure(.NO_NETWORK))
+						}
```

Comment:
Not every code path leads to a completion call here, could we add another kind of generic completion call for other failures?

Reply:
Ai good eye! I missed the completion call. 
Also I added VALUE_SET_CLIENT_ERROR so it is consistent to the other cases (ACCEPTANCE_RULE_CLIENT_ERROR and INVALIDATION_RULE_CLIENT_ERROR which are thrown for the default handler).